crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -194,32 +253,29 @@ impl CachingNativePriceEstimator {
             estimator,
             cache: Default::default(),
             high_priority: Default::default(),
-        });
-        let metrics = Metrics::instance(global_metrics::get_metric_storage_registry()).unwrap();
-        tokio::spawn(
-            update_recently_used_outdated_prices(
-                Arc::downgrade(&inner),
-                update_interval,
-                update_size,
-                max_age.saturating_sub(prefetch_time),
-                concurrent_requests,
-                metrics,
-            )
-            .instrument(tracing::info_span!("caching_native_price_estimator")),
-        );
-        Self {
-            inner,
             max_age,
-            metrics,
+        });
+
+        let update_task = UpdateTask {
+            inner: Arc::downgrade(&inner),
+            update_interval,
+            update_size,
+            prefetch_time,
+            concurrent_requests,
         }
+        .run()
+        .instrument(tracing::info_span!("caching_native_price_estimator"));
+        tokio::spawn(update_task);
```

Comment:
Should this PR be merged instead of https://github.com/cowprotocol/services/pull/1159? Asking because they seem incompatible.
Also in the other PR I found the separation of the `CachingNativePriceEstimator` and it's update task a bit nicer. By introducing a new function that spawns the update task you only had to pass the arguments related to the task if you actually intend to spawn the task. Don't get me wrong I still think it makes sense to have the update task for the `orderbook` but the separation of concerns is an added bonus IMO.